src/modules/navigator/takeoff.cpp

```diff
@@ -98,60 +98,31 @@ Takeoff::set_takeoff_position()
 {
 	struct position_setpoint_triplet_s *rep = _navigator->get_takeoff_triplet();
 
-	float abs_altitude = 0.0f;
+	float takeoff_altitude_amsl = 0.f;
 
-	float min_abs_altitude;
-
-	// TODO: review this, comments are talking about home pos, the validity is checked but the
-	// current altitude is used instead. Also, the "else" case does not consider the current altitude at all.
-	if (_navigator->home_alt_valid()) { //only use home position if it is valid
-		min_abs_altitude = _navigator->get_global_position()->alt + _navigator->get_takeoff_min_alt();
-
-	} else { //e.g. flow
-		min_abs_altitude = _navigator->get_takeoff_min_alt();
-	}
-
-	// Use altitude if it has been set. If home position is invalid use min_abs_altitude
-	events::LogLevel log_level = events::LogLevel::Disabled;
-
-	if (rep->current.valid && PX4_ISFINITE(rep->current.alt) && _navigator->home_alt_valid()) {
-		abs_altitude = rep->current.alt;
-
-		// If the altitude suggestion is lower than home + minimum clearance, raise it and complain.
-		if (abs_altitude < min_abs_altitude) {
-			if (abs_altitude < min_abs_altitude - 0.1f) { // don't complain if difference is smaller than 10cm
-				mavlink_log_critical(_navigator->get_mavlink_log_pub(),
-						     "Using minimum takeoff altitude: %.2f m\t", (double)_navigator->get_takeoff_min_alt());
-				log_level = events::LogLevel::Warning;
-			}
-
-			abs_altitude = min_abs_altitude;
-		}
+	if (rep->current.valid && PX4_ISFINITE(rep->current.alt)) {
+		takeoff_altitude_amsl = rep->current.alt;
 
 	} else {
-		// Use home + minimum clearance but only notify.
-		abs_altitude = min_abs_altitude;
+		takeoff_altitude_amsl = _navigator->get_global_position()->alt + _navigator->get_param_mis_takeoff_alt();
```

Comment:
I saw you have changed it to be above the current altitude. Theoretically, you could enter the mode, even when you are flying. I think in this case it would make sense to only use the home altitude (since you anyways wont decent and take at least the current altitude. If you are landed, it would make sense to take altitude from vehicle altitude.